src/cloudant/account.py

```diff
@@ -392,32 +392,34 @@ def _usage_endpoint(self, endpoint, year=None, month=None):
         optional year and month URL elements.
 
         :param str endpoint: Cloudant usage endpoint.
-        :param int year: Year to query against.  Defaults to None.
-        :param int month: Month to query against.  Defaults to None.
-        """
-        if year is not None:
-            endpoint = posixpath.join(endpoint, str(year))
-        if month is not None:
-            if year is None:
-                raise CloudantException(
-                    (
-                        "must supply both year and month "
-                        "to usage endpoint: {0}"
-                    ).format(endpoint)
-                )
-            endpoint = posixpath.join(endpoint, str(month))
+        :param int year: Year to query against.  Optional parameter.
+        Defaults to None.  If used, it must be accompanied by the ``month``.
+        :param int month: Month to query against. Optional parameter.
+        Defaults to None.  If used, it must be accompanied by the ``year``.
+        """
+        if (year is not None) and (month is not None):
+            endpoint = posixpath.join(endpoint, str(year), str(month))
```

Comment:
I dug into this a bit more...

I think if we change from using `posixpath.join` to using `urljoin` it should probably makes things clearer from a code readability standpoint and we'd then be using a library that is meant for all things URL related.  See [Py2 - urlparse.urljoin](https://docs.python.org/2/library/urlparse.html#urlparse.urljoin) and [Py3 - urllib.parse.urljoin](https://docs.python.org/3.2/library/urllib.parse.html#urllib.parse.urljoin).

I've added a separate issue #127 to track this, so that we can handle this at the repo level.